scripts/generate_simple_forced.py

```diff
@@ -0,0 +1,265 @@
+import sys
+
+sys.path.append("src")
+
+from pathlib import Path
+import xarray as xr
+import torch
+from jsonargparse import ArgumentParser
+
+from causaldynamics.creator import (
+    create_scm,
+    propagate_mlp,
+    logger,
+)
+from causaldynamics.data_io import (
+    create_output_dataset,
+    save_xr_dataset,
+    create_dynsys_dataset,
+)
+from causaldynamics.utils import process_confounders, check_confounders
+from causaldynamics.systems import (
+    solve_random_systems,
+    solve_system,
+    drive_linear,
+    drive_sin,
+)
+from causaldynamics.systems import get_adjacency_matrix_from_jac
+
+# Edge features
+_init_ratios = {
+    "nonlinear": [1.0, 0.0],
+    "periodic": [1.0, 1.0],
+}
+
+
+def generate(
+    *,
+    data_dir: str,
+    noise: float,
+    confounder: bool,
+    standardize: bool,
+    system_name: str,
+    seed: int,
+):
+    """
+    Generate all data, ranging from coupled models.
+
+    This function serves as the main entry point to generate dataset, including ground truth graphs.
+
+    Parameters
+    ----------
+    data_dir : str
+        Directory containing the time series data and adjacency matrices in a netCDF file.
+
+    noise : float
+        Amplitude of Langevin noise; >0 is an SDE.
+
+    confounder : bool
+        Whether to apply unobserved confounding post-processing.
+
+    standardize : bool
+        Whether to standardize the trajectories following the iSCM algorithm.
+
+    system_name : str
+        The system representing each node e.g., Lorenz84 or random.
+
+    seed : int
+        Index representing SCM that is randomly constructed.
+
+    Returns
+    -------
+    None
+        Results are saved to disk at data_dir/output/coupled/<experiment_id>/<system_name>.nc as NetCDF files
+
+    Notes
+    -----
+    Example usage:
+        python scripts/generate_simple_forced.py \
+                                            --data_dir data \
+                                            --noise 0.5 \
+                                            --confounder \
+                                            --standardize \
+                                            --system_name random \
+                                            --seed 0
```

Comment:
@juannat7 

I implemented this script to generate simple systems with three forcing options: `linear`, `periodic` and `noforcing`. The trajectory of the dynamical system is exactly the same such that we have good comparison between these forcing options and we can investigate whether confounded system forcing impacts the causal discovery algorithms. What do you think?